### PR TITLE
[RFR] Implement text input types

### PR DESCRIPTION
**Pull request by jpetitcolas**
- [x] Allow to customize `<TextInput />` fields
- [x] Display differently emails and URLs depending of `<TextField />` type

### CODE REVIEW

**Comment by jpetitcolas** [comment]
Moving the boolean field into another PR. Switching to RFR.

File: docs/InputsFields.md [hunk 1]
```diff
@@ -10,12 +10,50 @@ display view of a record property.
...
+* **label:** text indicating which piece of data should be filled,
 * **record:** object describing your entity as retrieved by your API,
 * **source:** property name of your entity to view/edit.
 
 In addition to these common `props`, you can also configure each field independently,
 depending of their types.
 
+## Text Fields and Inputs (and Related Ones: Email, Password, etc.)
+
+## Text Inputs
```
Comment (Kmaschta) [inline docs/InputsFields.md]: s/##/###/

File: docs/InputsFields.md [hunk 2]
```diff
@@ -10,12 +10,50 @@ display view of a record property.
...
+
+<TextInput
+    label="Email Address"
+    source="email"
+    type="email" />
+```
+
+By default, it renders a `text` field.
+
+## Text Fields
```
Comment (Kmaschta) [inline docs/InputsFields.md]: s/##/###/

File: src/mui/field/TextField.js [hunk 3]
```diff
@@ -1,11 +1,26 @@
 import React, { PropTypes } from 'react';
 
-const TextField = ({ source, record = {} }) => <span>{record[source]}</span>;
+const TextField = ({ source, record = {}, type }) => {
```
Comment (fzaninotto) [inline src/mui/field/TextField.js]: I'd prefer if we bundled separate components for each of these cases.

File: src/mui/input/DisabledInput.js [hunk 4]
```diff
@@ -14,7 +14,6 @@ DisabledInput.propTypes = {
 };
 
 DisabledInput.defaultProps = {
-    includesLabel: true,
```
Comment (fzaninotto) [inline src/mui/input/DisabledInput.js]: why? It should include label

**Comment by jpetitcolas** [comment]
Code review applied.

**Pull request merged**